bioio_base/reader.py

```diff
@@ -217,6 +218,26 @@ def current_resolution_level(self) -> int:
         """
         return self._current_resolution_level
 
+    @property
+    def resolution_level_dims(self) -> Dict[int, Tuple[int, ...]]:
+        """
+        Returns
+        -------
+        resolution_level_dims: Dict[int, Tuple[int, ...]]
+            resolution level dictionary of shapes.
+        """
+        if self._resolution_level_dict is None:
+            initial_resoluiton_level = self.current_resolution_level
+            resolution_level_dict = {}
+
+            for level in self.resolution_levels:
+                self.set_resolution_level(level)
+                resolution_level_dict[level] = self.shape
+            self._resolution_level_dict = resolution_level_dict
```

Comment:
If this is cached in the Reader object, `_reset_self` should delete it for proper interaction with `set_scene`.

Reply:
Good catch.  Definitely agree.